### ISSUE TITLE
Add rule to check uniqueness of `operationId`

### ISSUE THREAD

**Issue by magicmatatjahu** [stoplightio/spectral#2102]
**User story.**
As a user, I want to know if the operations defined in my AsyncAPI document have a unique `operationId`.

**Describe the solution you'd like**
The following AsyncAPI document is invalid and Spectral should throw errors due to non-uniqueness of `operationId`.

```yaml
asyncapi: '2.3.0'
...

channels:
  someChannel:
    publish:
      operationId: someId
      ...
    subscribe:
      operationId: someId
      ...
```

**Additional context**
Part of #2100
